reactive/kotlinx-coroutines-rx2/src/RxChannel.kt

```diff
@@ -31,22 +31,24 @@ public suspend inline fun <T> ObservableSource<T>.collect(action: (T) -> Unit):
     toChannel().consumeEach(action)
 
 @PublishedApi
+@Suppress("UNCHECKED_CAST")
 internal fun <T> MaybeSource<T>.toChannel(): ReceiveChannel<T> {
     val channel = SubscriptionChannel<T>()
-    subscribe(channel)
+    (this as MaybeSource<T & Any>).subscribe(channel)
     return channel
 }
 
 @PublishedApi
+@Suppress("UNCHECKED_CAST")
 internal fun <T> ObservableSource<T>.toChannel(): ReceiveChannel<T> {
     val channel = SubscriptionChannel<T>()
-    subscribe(channel)
+    (this as ObservableSource<T & Any>).subscribe(channel)
     return channel
 }
 
 @Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
 private class SubscriptionChannel<T> :
-    LinkedListChannel<T>(null), Observer<T>, MaybeObserver<T>
+    LinkedListChannel<T>(null), Observer<T & Any>, MaybeObserver<T & Any>
```

Comment:
Ditto: the casts above will not be needed if these bounds are removed.

Reply:
But this change *is* required for Rx3. Shouldn't `RxChannel` be changed for rx3 as well?

Reply:
Also, rx3's `RxMaybe`, too, gives me warnings.